base/abstractarray.jl

```diff
@@ -1237,15 +1238,15 @@ oneunit(x::AbstractMatrix{T}) where {T} = _one(oneunit(T), x)
 iterate_starting_state(A) = iterate_starting_state(A, IndexStyle(A))
 iterate_starting_state(A, ::IndexLinear) = firstindex(A)
 iterate_starting_state(A, ::IndexStyle) = (eachindex(A),)
-iterate(A::AbstractArray, state = iterate_starting_state(A)) = _iterate(A, state)
+@inline iterate(A::AbstractArray, state = iterate_starting_state(A)) = _iterate(A, state)
```

Comment:
I'm not entirely sure about adding this `@inline`, but it was used in `Array` and `Memory` methods already.